src/h5web/visualizer/utils.ts

```diff
@@ -11,24 +15,34 @@ function removeRedundantVis(visArr: Vis[]): Vis[] {
   return visArr;
 }
 
-export function getSupportedVis(
-  entity: Entity | undefined
-): { supportedVis: Vis[]; error?: Error } {
-  if (!entity) {
-    return { supportedVis: [] };
-  }
+export function getSupportedVis(entity: Entity): Vis[] {
+  const supportedVis = Object.entries(VIS_DEFS).reduce<Vis[]>(
+    (arr, visDefEntry) => {
+      const [vis, { supportsEntity }] = visDefEntry;
+      return supportsEntity(entity) ? [...arr, vis as Vis] : arr;
+    },
+    []
+  );
+
+  return removeRedundantVis(supportedVis);
+}
 
-  try {
-    const supportedVis = Object.entries(VIS_DEFS).reduce<Vis[]>(
-      (arr, visDefEntry) => {
-        const [vis, { supportsEntity }] = visDefEntry;
-        return supportsEntity(entity) ? [...arr, vis as Vis] : arr;
-      },
-      []
-    );
-
-    return { supportedVis: removeRedundantVis(supportedVis) };
-  } catch (error) {
-    return { supportedVis: [], error };
+export function getDefaultEntity(
+  entity: Entity,
+  entitiesStore: FetchStore<Entity, string>
```

Comment:
I couldn't make it a hook and use `useContext` inside because of the recursion (the recursive call comes after an `if`/`return`, which breaks a rule of hooks).